dvc/output/base.py

```diff
@@ -248,14 +267,22 @@ def changed(self):
         logger.debug(str(status))
         return bool(status)
 
+    @property
+    def dvcignore(self):
+        return None
+
     @property
     def is_empty(self):
         return self.fs.is_empty(self.path_info)
 
     def isdir(self):
+        if self._check_path_dvcignore(self.path_info):
+            return False
         return self.fs.isdir(self.path_info)
```

Comment:
@efiop , `dvcignore` didn't test where the path is a file or a dir. The only difference of dirs and files are the matching of pattern like `paths/`.

Reply:
@karajan1001 Right, I'm saying that you are using `is_ignored` which will interpret `path` as a local path and will try to use `os.path.isdir` https://github.com/iterative/dvc/blob/e7571ab3251ef4417fa196e7ad45b7f6f8a42ee3/dvc/ignore.py#L354 on it, while it should use `fs.isdir`, because `fs` might be a GitFileSystem, for example.

Reply:
In general, right now `dvcignore` should only work for `fs`(and `path`) that only belong to `dvcignore.fs`(filesystem that dvcignore was collected from).